### ISSUE TITLE
TimeoutError: context deadline exceeded

### ISSUE THREAD

**Issue by christroutner** [orbitdb/orbit-db#946]
I'm trying to instantiate a keyvalue database that I've opened before. The only things I've changed recently is that I updated IPFS and orbit-db. I haven't changed the code from what was working before.

```javascript
this.db = await orbitdb.keyvalue(dbName, options)
```

And I'm getting this error:

```
Error in createDb():  TimeoutError: context deadline exceeded
    at Object.errorHandler [as handleError] (/home/safeuser/ipfs-p2wdb-service/node_modules/ipfs-http-client/cjs/src/lib/core.js:90:15)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Client.fetch (/home/safeuser/ipfs-p2wdb-service/node_modules/ipfs-utils/src/http.js:145:9)
    at async Object.get (/home/safeuser/ipfs-p2wdb-service/node_modules/ipfs-http-client/cjs/src/block/get.js:10:17)
    at async Object.read (/home/safeuser/ipfs-p2wdb-service/node_modules/orbit-db-io/index.js:59:17)
    at async OrbitDB.open (/home/safeuser/ipfs-p2wdb-service/node_modules/orbit-db/src/OrbitDB.js:452:22)
    at async OrbitDBAdapter.createDb (/home/safeuser/ipfs-p2wdb-service/src/adapters/orbit/index.js:91:17)
    at async OrbitDBAdapter.start (/home/safeuser/ipfs-p2wdb-service/src/adapters/orbit/index.js:49:9)
    at async P2WDB.start (/home/safeuser/ipfs-p2wdb-service/src/adapters/p2wdb/index.js:57:7)
    at async Adapters.start (/home/safeuser/ipfs-p2wdb-service/src/adapters/index.js:59:7)
    at async Controllers.attachControllers (/home/safeuser/ipfs-p2wdb-service/src/controllers/index.js:42:5)
    at async Server.startServer (/home/safeuser/ipfs-p2wdb-service/bin/server.js:95:7)

```

I'm using the following versions:
- go-ipfs: 0.11.0
- node.js: 14.18.1
- npm: 8.1.3
- ipfs-http-client: 55.0.0
- orbit-db: 0.28.0

**Comment by christroutner** [comment]
I think Issue #919 is related. The timeout appears to be due to the inability to fetch the database manifest.

What is the proper way to handle this kind of error? 
- Should I put it in a loop and try/catch statement, and just keep trying to open the database until it succeeds?
- Is there a way to create a new manifest if one can't be fetched?

**Comment by tabcat** [comment]
> Should I put it in a loop and try/catch statement, and just keep trying to open the database until it succeeds?

up to you, you may also be able to change the timeout from the default https://github.com/orbitdb/orbit-db/blob/2b6b267285bb6a9e8bbf835857423753b049fa5f/src/OrbitDB.js#L452

> Is there a way to create a new manifest if one can't be fetched?

you can create a new manifest with determineAddress and supplying the necessary options


sorry the error wasnt more helpful and i think the timeout option will need to be documented as well

**Comment by christroutner** [comment]
Thanks for the advice @tabcat!

I built a small test script to look into this issue and here is what I found:

- I had no issue creating a new keyvalue database.
- My app was trying to open a pre-existing database with a full hash name: `/orbitdb/zdpuAvdpQuBXR9o2xM2W4wr6dws2zzWoSwEUMRdHaJjcagBCX/psf-bch-p2wdb-keyvalue-v1.0.0-0004`
- The root-cause for why the manifest timed-out was because I had switched my IPFS node to a private network. It could therefore not connect to any nodes that had the manifest for the pre-existing database.

The solution for me was to create a new database and bootstrap my p2p network again, with the new database.
